From 2e4be87bfec8610752ed715279d732679a1de37b Mon Sep 17 00:00:00 2001
Message-Id: <2e4be87bfec8610752ed715279d732679a1de37b.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: "Thulasimani, Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 28 Jan 2015 21:01:01 +0530
Subject: [PATCH 50/95] MUST_REBASE [VPG]: drm/i915: Wrapper function for PPS
 delays

This patch retrieves the pps delays parsed from vbt in i915
for use inside ADF driver.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5485
Change-Id: I60e5bc10ef450eac56c1e4f1040e49c37e383062
Signed-off-by: Thulasimani, Sivakumar <sivakumar.thulasimani@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    5 +++++
 include/drm/i915_adf.h                  |    1 +
 2 files changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 8883d2b..80351d2 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -204,4 +204,9 @@ void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num)
 }
 EXPORT_SYMBOL(intel_get_vbt_disp_conf);
 
+unsigned short *intel_get_vbt_pps_delays(void)
+{
+	return (unsigned short *)&i915_adf_dev->vbt.edp_pps;
+}
+EXPORT_SYMBOL(intel_get_vbt_pps_delays);
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 2709773..9beb1ad 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -60,6 +60,7 @@ extern u8 intel_adf_get_platform_id(void);
 extern u16 intel_get_dsi_port_frm_vbt(void);
 extern void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num);
 extern struct i2c_adapter *intel_adf_get_gmbus_adapter(u8 port);
+extern unsigned short *intel_get_vbt_pps_delays(void);
 #endif
 
 
-- 
1.7.9.5

